consensus/polybft/proposer_calculator_test.go

```diff
@@ -0,0 +1,510 @@
+package polybft
+
+import (
+	"bytes"
+	"math"
+	"testing"
+
+	bls "github.com/0xPolygon/polygon-edge/consensus/polybft/signer"
+	"github.com/0xPolygon/polygon-edge/types"
+	"github.com/hashicorp/go-hclog"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func TestProposerCalculator_SetIndex(t *testing.T) {
+	t.Parallel()
+
+	validators := newTestValidatorsWithAliases([]string{"A", "B", "C", "D", "E"}, []uint64{10, 100, 1, 50, 30})
+	metadata := validators.getPublicIdentities()
+
+	vs, err := validators.toValidatorSet()
+	require.NoError(t, err)
+
+	snapshot := NewProposerSnapshot(1, metadata)
+	pc := NewProposerCalculatorFromSnapshot(snapshot, hclog.NewNullLogger())
+
+	// validate no changes to validator set positions
+	for i, v := range vs.Accounts() {
+		assert.Equal(t, metadata[i].Address, v.Address)
+	}
+
+	proposer, err := pc.CalcProposer(0, 1)
+	require.NoError(t, err)
+	assert.Equal(t, proposer, metadata[1].Address)
+	// validate no changes to validator set positions
+	for i, v := range vs.Accounts() {
+		assert.Equal(t, metadata[i].Address, v.Address)
+	}
+}
+
+func TestProposerCalculator_CalculateProposer(t *testing.T) {
+	t.Parallel()
+
+	validators := newTestValidatorsWithAliases([]string{"A", "B", "C", "D", "E"}, []uint64{1, 2, 3, 4, 5})
+	metadata := validators.getPublicIdentities()
+
+	vs, err := validators.toValidatorSet()
+	require.NoError(t, err)
+
+	snapshot := NewProposerSnapshot(0, metadata)
+	pc := NewProposerCalculatorFromSnapshot(snapshot, hclog.NewNullLogger())
+
+	assert.Equal(t, int64(15), vs.totalVotingPower)
+
+	currProposerAddress, err := pc.CalcProposer(0, 0)
+	require.NoError(t, err)
+	assert.Equal(t, metadata[4].Address, currProposerAddress)
+
+	proposerAddressR1, err := pc.CalcProposer(1, 0)
+	require.NoError(t, err)
+	assert.Equal(t, metadata[3].Address, proposerAddressR1)
+
+	proposerAddressR2, err := pc.CalcProposer(2, 0)
+	require.NoError(t, err)
+	assert.Equal(t, metadata[2].Address, proposerAddressR2)
+
+	proposerAddressR3, err := pc.CalcProposer(3, 0)
+	require.NoError(t, err)
+	assert.Equal(t, metadata[1].Address, proposerAddressR3)
+
+	proposerAddressR4, err := pc.CalcProposer(4, 0)
+	require.NoError(t, err)
+	assert.Equal(t, metadata[4].Address, proposerAddressR4)
+
+	proposerAddressR5, err := pc.CalcProposer(5, 0)
+	require.NoError(t, err)
+	assert.Equal(t, metadata[3].Address, proposerAddressR5)
+
+	proposerAddressR6, err := pc.CalcProposer(6, 0)
+	require.NoError(t, err)
+	assert.Equal(t, metadata[0].Address, proposerAddressR6)
+}
+
+func TestProposerCalculator_SamePriority(t *testing.T) {
+	t.Parallel()
+
+	keys, err := bls.CreateRandomBlsKeys(5)
```

Comment:
Can you use `newTestValidatorsWithAliases` in this scenario?

Reply:
It looks like that all the proposer_calculator tests should be switched to `TestValidatorsWithAliases`. Furthermore, some of the tests will be changed by https://github.com/0xPolygon/polygon-edge/pull/1048 which should be rebased on top of this PR. 
Is it ok to modify  the tests after all the PRs related to `VotingPower` are merged?

Reply:
Sure, please add to Jira.

Reply:
Here is the task: https://polygon.atlassian.net/browse/EVM-329